modules/php/Actions/Scenario4/CirclePlantOrWater.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Bga\Games\WelcomeToTheMoon\Actions\Scenario4;
+
+use Bga\Games\WelcomeToTheMoon\Core\Notifications;
+use Bga\Games\WelcomeToTheMoon\Managers\Players;
+use Bga\Games\WelcomeToTheMoon\Models\Player;
+
+class CirclePlantOrWater extends \Bga\Games\WelcomeToTheMoon\Models\Action
```

Comment:
Why not making this extend GenericPickSlot ?
This will directly handle checking if the slots are both empty or not, which I think it actually missing below and wont require any additional js

Reply:
Done